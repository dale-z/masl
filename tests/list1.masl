[int] list = [int]{1, 2, 3};
printInt(list:.get(0));
printChar('\n');
list:.set(0, 0);
printInt(list:.get(0));
printChar('\n');
list:.insert(0, -1);
printInt(list:.get(0));
printChar('\n');
list:.remove(1);
printInt(list:.get(1));
printChar('\n');
list:.append(10);
printInt(list:.get(list:.size()-1));
printChar('\n');
list:.set(0,-3);
printInt(list:[0]);
printChar('\n');

bool isEvenNum(int num) { return (num%2 == 0); }

[int] list2 = [int]{1, 2, 3, 4, 5, 6};
[int] list3 = list2:.filter(isEvenNum);
for(int i : list3) {
	printInt(i);
}
printChar('\n');
printInt(list2:.count(isEvenNum));
