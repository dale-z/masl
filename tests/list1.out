1
0
-1
2
10
-3
246
3